src/base/settingvalue.h

```diff
@@ -68,11 +69,45 @@ class CachedSettingValue
     CachedSettingValue<T> &operator=(const T &newValue)
     {
         m_value = newValue;
-        SettingsStorage::instance()->storeValue(m_keyName, m_value);
+        storeValue(m_value);
         return *this;
     }
 
 private:
+    // regular load/save pair
+    template <typename U, typename std::enable_if<!std::is_enum<U>::value, int>::type = 0>
+    U loadValue(const U &defaultValue)
+    {
+        return SettingsStorage::instance()->loadValue(m_keyName, defaultValue).template value<T>();
+    }
+
+    template <typename U, typename std::enable_if<!std::is_enum<U>::value, int>::type = 0>
+    void storeValue(const U &value)
+    {
+        SettingsStorage::instance()->storeValue(m_keyName, value);
+    }
+
+    // load/save pair for an enum
+    // saves literal value of the enum constant, obtained from QMetaEnum
```

Comment:
@sledgehammer999:  
I believe this code comment explains the purpose of this PR.